bundles/org.openhab.binding.goecharger/src/main/java/org/openhab/binding/goecharger/internal/handler/GoEChargerHandler.java

```diff
@@ -341,96 +284,78 @@ public void handleCommand(ChannelUID channelUID, Command command) {
 
     @Override
     public void initialize() {
-        config = getConfigAs(GoEChargerConfiguration.class);
-        allChannels = getThing().getChannels().stream().map(channel -> channel.getUID().getId())
-                .collect(Collectors.toList());
-
-        updateStatus(ThingStatus.UNKNOWN);
+        super.initialize();
+    }
 
-        startAutomaticRefresh();
-        logger.debug("Finished initializing!");
+    private String getReadUrl() {
+        return GoEChargerBindingConstants.API_URL.replace("%IP%", config.ip.toString());
     }
 
-    private String getUrl(String type) {
-        return type.replace("%IP%", StringUtils.trimToEmpty(config.ip));
+    private String getWriteUrl(String key, String value) {
+        return GoEChargerBindingConstants.MQTT_URL.replace("%IP%", config.ip.toString()).replace("%KEY%", key)
+                .replace("%VALUE%", value);
     }
 
     private void sendData(String key, String value) {
-        String urlStr = getUrl(GoEChargerBindingConstants.MQTT_URL).replace("%KEY%", key).replace("%VALUE%", value);
-        logger.debug("POST URL = {}", urlStr);
+        String urlStr = getWriteUrl(key, value);
+        logger.trace("POST URL = {}", urlStr);
 
         try {
-            ContentResponse contentResponse = httpClient.newRequest(urlStr).method(HttpMethod.POST)
+            HttpMethod httpMethod = HttpMethod.POST;
+            ContentResponse contentResponse = httpClient.newRequest(urlStr).method(httpMethod)
                     .timeout(5, TimeUnit.SECONDS).send();
             String response = contentResponse.getContentAsString();
-            logger.debug("POST Response: {}", response);
-            GoEStatusResponseDTO result = gson.fromJson(response, GoEStatusResponseDTO.class);
-            updateChannelsAndStatus(result, null);
+
+            logger.trace("{} Response: {}", httpMethod.toString(), response);
+
+            var statusCode = contentResponse.getStatus();
+            if (!(statusCode == 200 || statusCode == 204)) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Request response was unsuccessful");
+                logger.debug("Could not send data, Response {}, StatusCode: {}", response, statusCode);
+            }
         } catch (InterruptedException | TimeoutException | ExecutionException | JsonSyntaxException e) {
-            updateChannelsAndStatus(null, e.getMessage());
```

Comment:
For `InterruptedException`, please call `Thread.currentThread().interrupt()`. See #12365.